spring-cql/src/test/java/org/springframework/cassandra/test/integration/core/cql/generator/CqlIndexSpecificationAssertions.java

```diff
@@ -19,25 +19,45 @@
 
 import org.springframework.cassandra.core.keyspace.IndexDescriptor;
 
-import com.datastax.driver.core.ColumnMetadata.IndexMetadata;
+import com.datastax.driver.core.IndexMetadata;
 import com.datastax.driver.core.Session;
+import com.datastax.driver.core.TableMetadata;
 
 /**
  * @author David Webb
  * @author Matthew T. Adams
+ * @author Antoine Toulme
  */
 public class CqlIndexSpecificationAssertions {
 
+	/**
+	 * Assert the existence of an index using the index name.
+	 *
+	 * @param expected
+	 * @param keyspace
+	 * @param session
+	 */
 	public static void assertIndex(IndexDescriptor expected, String keyspace, Session session) {
-		IndexMetadata imd = session.getCluster().getMetadata().getKeyspace(keyspace.toLowerCase())
-				.getTable(expected.getTableName().toCql()).getColumn(expected.getColumnName().toCql()).getIndex();
+		TableMetadata tableMetadata = session.getCluster().getMetadata().getKeyspace(keyspace.toLowerCase())
+				.getTable(expected.getTableName().toCql());
 
-		assertEquals(expected.getName(), imd.getName());
+		IndexMetadata imd = tableMetadata.getIndex(expected.getName().toCql());
+
+		assertEquals(expected.getName(), imd == null ? null : imd.getName());
```

Comment:
`assertThat(imd != null ? imd.getName() : null, is(equalTo(expected.getName()));`